staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -457,6 +457,14 @@ func (f *DeltaFIFO) queueActionLocked(actionType DeltaType, obj interface{}) err
 		if err != nil {
 			return err
 		}
+		// If the returned object is nil, storing a nil value can cause panics
+		// if consumers assume the object will be always not nil:
+		// > panic: interface conversion: interface {} is nil, not ...
+		// Users can use the tranformer to not store objects in memory
+		// by returning nil for those objects they want to filter.
+		if obj == nil {
+			return nil
+		}
```

Comment:
Hmm - storing a nil seems problematic, but not storing anything is also problematic, because that means that e.g. every update will be treated as creation...

I'm wondering if we shouldn't return an error instead? Or store some guard or sth like that?

Reply:
making this a no-op isn't correct, depending on the action, right? a nil on add could maybe be a no-op... a nil on update should turn into a remove, right? and a nil on delete is just... weird... shouldn't that still be a delete?

Reply:
I couldn't find a good answer so I suggested to make return nil mean "I want to filter"  since the transformer does not have any idea of the actions, only of the object, so it will always apply the same logic to any action

Reply:
What does "I want to filter" exactly mean? Do you rather mean: "Do not store" ?

Reply:
what do we want to do with this? we let panic?